apps/web-giddh/src/app/services/general.service.ts

```diff
@@ -1693,25 +1694,252 @@ export class GeneralService {
     }
 
     /**
-     * This will return the system current user time zone
+     * Retrieves the operating system configuration based on the user agent string.
+     *
+     * @returns {string} The name of the operating system.
+     * @memberof GeneralService
+     */
+    public getOsConfiguration() {
+        const userAgent = window.navigator.userAgent;
+        let osName;
+
+        if (userAgent.indexOf("Win") != -1) {
+            osName = "Windows";
+        } else if (userAgent.indexOf("Mac") != -1) {
+            osName = "Macintosh";
+        } else if (userAgent.indexOf("Linux") != -1) {
+            osName = "Linux";
+        } else if (userAgent.indexOf("Android") != -1) {
+            osName = "Android";
+        } else if (userAgent.indexOf("iOS") != -1) {
+            osName = "iOS";
+        } else {
+            osName = "Unknown";
+        }
+
+        return osName;
+    }
+
+    /**
+     * Retrieves the device manufacturer based on the user agent string.
+     *
+     * @returns {string} The device manufacturer.
+     * @memberof GeneralService
+     */
+    public getDeviceManufacture() {
+        const userAgent = window.navigator.userAgent;
+        let deviceManufacture = 'Unknown';
+
+        if (userAgent.indexOf('iPhone') !== -1 || userAgent.indexOf('iPad') !== -1) {
+            deviceManufacture = 'Apple';
+        } else if (userAgent.indexOf('Android') !== -1) {
+            deviceManufacture = 'Samsung'; // Assuming Samsung for Android, could be any Android device manufacturer
+        } // Add additional checks for other common devices if needed
+
+        return deviceManufacture;
+    }
+
+    /**
+     * Retrieves the current timestamp in ISO format.
+     *
+     * @returns {string} The current timestamp.
+     * @memberof GeneralService
+     */
+    public getTimesStamp() {
```

Comment:
public getTimesStamp(): string {

add return type string or any which is better